dali/dfu/dfuserver.cpp

```diff
@@ -113,48 +127,56 @@ int main(int argc, const char *argv[])
 
     NoQuickEditSection xxx;
 
-    Owned<IFile> file = createIFile("dfuserver.xml");
-    if (file->exists())
-        globals.setown(createPTreeFromXMLFile("dfuserver.xml", ipt_caseInsensitive));
-    else
-        globals.setown(readOldIni());
-
-    for (unsigned i=1;i<(unsigned)argc;i++) {
-        const char *arg = argv[i];
-        StringBuffer prop("@");
-        StringBuffer val;
-        while (*arg && *arg != '=')
-            prop.append(*arg++);
-        if (*arg) {
-            arg++;
-            while (isspace(*arg))
-                arg++;
-            val.append(arg);
-            prop.clip();
-            val.clip();
-            if (prop.length()>1)
-                globals->setProp(prop.str(), val.str());
-        }
+    try
+    {
+        globals.setown(loadConfiguration(defaultYaml, argv, "dfuserver", "DFUSERVER", "dfuserver.xml", nullptr));
+    }
+    catch (IException * e)
+    {
+        UERRLOG(e);
+        e->Release();
+        return 1;
+    }
+    catch(...)
+    {
+        OERRLOG("Failed to load configuration");
+        return 1;
     }
+
     StringBuffer daliServer;
     StringBuffer queue;
-    if (!globals->getProp("@DALISERVERS", daliServer)||!globals->getProp("@QUEUE", queue)) {
-        usage();
-        globals.clear();
-        releaseAtoms();
+    if (!globals->getProp("@daliServers", daliServer))
+        globals->getProp("@DALISERVERS", daliServer);
+    if (0==daliServer.length())
+    {
+        PROGLOG("daliservers not specified in dfuserver.xml");
+        return 1;
+    }
+    if (!globals->getProp("@QUEUE", queue))
+        globals->getProp("@queue", queue);
```

Comment:
In container mode, I don't think we should have a configurable "queue" option.
It should be implicit based on name of the dfuserver instance, as it is for other engines/agents etc.
There are utility function in daqueue.hpp, a new one could be added for getDfuServerQueueName

Reply:
Is there going to be a dfuserver instance per active job?
With replicas auto scaled, based on active/queue size? (maybe at some point).

I think until then, we need an option that controls max number of jobs it's going to run in parallel.
At the moment, there's some very odd code https://github.com/hpcc-systems/HPCC-Platform/blob/ff5e5f118e9d77de533725ce86914f4c66339ecf/dali/dfu/dfuserver.cpp#L192, that determines how many jobs it can run in parallel, by seeing how many comma separated queues are defined in @queues, then creating that many listeners.
NB: what our ops tend to do is set bare-metal @queue to e.g. "dfuserver_queue,dfuserver_queue,dfuserver_queue,dfuserver_queue,dfuserver_queue"
which in effect causes it to run up to 5 jobs at one time.

For now, I'd suggest adding a maxJobs config option (similar to other components), and creating that many listeners.